source/fundamentals/monitoring/connection-monitoring.txt

```diff
@@ -0,0 +1,321 @@
+.. _golang-connection-monitoring:
+
+=====================
+Connection Monitoring
+=====================
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: code example, performance, monitor
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecols
+
+Overview
+--------
+
+This guide shows you how to use the {+driver-short+} to monitor the
+driver's connection pool. A connection pool is a set of open
+Transmission Control Protocol (TCP) connections that your driver
+maintains with a MongoDB instance. Connection pools help reduce the
+number of new connections your application needs to create,
+which might make your application run faster.
+
+You can use information about changes to your connection pool in your
+application, or you can monitor the connection pool to learn more about
+how the driver uses connections.
+
+Subscribe to Events
+-------------------
+
+You can access details about connection pool events by subscribing to them
+in your application. The following example demonstrates how to subscribe
+to the ``PoolEvent`` event by instantiating a
+``PoolMonitor`` and connecting to a deployment:
+
+.. code-block:: go
+
+   var eventArray []*event.PoolEvent
+   cxnMonitor := &event.PoolMonitor{
+       Started: func(e *event.PoolEvent) {
+           eventArray = append(eventArray, e)
+       },
+   }
+   clientOpts := options.Client().ApplyURI(uri).SetPoolMonitor(cxnMonitor)
+   client, err := mongo.Connect(clientOpts)
+
+Event Descriptions
+------------------
+
+The following table describes the types of pool events that the driver
+emits:
+
+.. list-table::
+   :widths: 33 67
+   :header-rows: 1
+
+   * - Pool Event Type
+     - Description
+
+   * - ``ConnectionPoolCreated``
+     - Created when a connection pool is created.
+
+   * - ``ConnectionPoolReady``
+     - Created when a connection pool is ready.
+
+   * - ``ConnectionPoolCleared``
+     - Created when a connection pool is cleared.
```

Comment:
I'm guessing this means all operations have been completed/the pools is empty - could you clarify this?